### ISSUE TITLE
[BUG] Exported OPML does not validate

### ISSUE THREAD

**Issue by gingerbeardman** [FreshRSS/FreshRSS#6140]
**Describe the bug**
OPML does not validate and fails to be imported into some other apps and services

**To Reproduce**
1. Export OPML from FreshRSS
2. Try to import OPML in [Bulletin RSS app](https://apps.apple.com/gb/app/bulletin-ai-news/id6476572500) (fails)
3. Validate OPML (fails)

**Expected behavior**
- OPML should validate

**Screenshots**
- <img width="400" alt="Screen shot 2024-03-01 at 15 27 22" src="https://github.com/FreshRSS/FreshRSS/assets/49612/067be088-e291-471d-8359-68316db88853">

**Environment information (please complete the following information):**
- Device: Synology
- OS: DSM 6.2.3
- Browser: Safari 17.3.1 (macOS 14.3.1)
- FreshRSS version: 1.23.1
- Database version: PDO?
- PHP version: 8.3.3
- Installation type: docker

**Additional context**
- Validator http://scripting.com/code/opmlvalidator/
- [Hosted OPML URL](https://gist.githubusercontent.com/gingerbeardman/ed7a2d03f2cf7af7d6c85397db07e420/raw/2d135ce9180d7ab14c399203b444a0d3ce245e05/freshrss.opml.xml)
- [Archived OPML ZIP](https://github.com/FreshRSS/FreshRSS/files/14462898/feeds_2024-03-01.opml.xml.zip)

**Comment by jsspen** [comment]
Running into the same issue...

**Comment by Alkarex** [comment]
[That validator ](http://scripting.com/code/opmlvalidator/) seems quite buggy. It does not even work on the official OPML examples:
http://opml.org/spec2.opml#1629043023000

![image](https://github.com/FreshRSS/FreshRSS/assets/1008324/ff44f103-a788-4f8b-988f-0a7493626fc4)

One line with a supposed error is the following:

```xml
<outline text="Ink &amp; Switch" type="rss"
  xmlUrl="https://www.inkandswitch.com/index.xml"
  htmlUrl="https://www.inkandswitch.com/"
  description="Recent content on Ink &amp; Switch"/>
```

But this line is valid. It is valid XML, and also valid OPML according to the specification http://opml.org/spec2.opml#1629042276000 :

> Text attributes may contain encoded HTML markup

It would be wrong to perform a double-encoding, as this would break other compliant tools.

Regarding errors like `An element should only have known attributes. Unknown: frss:filtersActionRead`, this is also a bug from the validator, according to the specification http://opml.org/spec2.opml#1629042978000

> an OPML file may contain elements and attributes not described on this page, only if those elements are defined in a namespace

And finally for errors like `The type attribute on an element should be a known type`, yet another validator bug, according to http://opml.org/spec2.opml#1629042978000

> OPML can also be extended by the addition of new values for the type attribute.

In conclusion, as far as I can see, OPML export from FreshRSS is exactly as it should. However, if someone finds any actual deviation from the OPML specification, do report it.
As for this ticket, I believe the problem should be reported to the Bulletin RSS app and to that validator.

At a quick glance, the logic of the validator seems bogus, as it performs an [XML parsing](https://github.com/scripting/opml.org/blob/5ea01b2f0fd52867c1b476dc0fff1aa6f209a13d/opmlValidator/code.js#L272) (which is good) but then does some manual check of the [XML entities](https://github.com/scripting/opml.org/blob/5ea01b2f0fd52867c1b476dc0fff1aa6f209a13d/opmlValidator/code.js#L129-L130), forgetting that those have already been decoded by the XML parser...

**Comment by Alkarex** [comment]
Upstream bug report https://github.com/scripting/opml.org/issues/17

**Comment by gingerbeardman** [comment]
So I'm wondering why the FreshRSS OPML can't be imported into other apps? Are they also using buggy code? Surely every app can't be using buggy OPML parsing?

**Comment by Frenzie** [comment]
Your sample OPML export works in Gnome Feeds and I'd be surprised if it didn't work in QuiteRSS but I can't install that right now.

**Comment by gingerbeardman** [comment]
Thanks! Will check more apps. Super bad luck on my part that the ones I've tried failed to import.

**Comment by Frenzie** [comment]
If there's more than one that fails there's a minor chance that it'd be interesting to learn if there's something common they balk on, but it's quite simple, well-formed XML and valid OPML.

If I had to guess, my bet would be on them not being able to deal with `frss:xxx`, even though that's standard namespace stuff. (I've dealt with and on occasion even half created incomplete XML parsers in the past, without proper namespace support. However, they still wouldn't have failed on these but simply ignored them as unknown attributes.)

**Comment by Alkarex** [comment]
The first bug is fixed in the validator https://github.com/scripting/opml.org/issues/17

**Comment by Alkarex** [comment]
Second bug out of three reported: https://github.com/scripting/opml.org/issues/18

**Comment by Alkarex** [comment]
Third bug reported: https://github.com/scripting/opml.org/issues/19
